### PR TITLE
Enable td-agent in Mariner

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Enable td-agent support in Mariner

###### Change Log  

- Initial version Mariner

###### Does this affect the toolchain?  

**YES**
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Local build

### CODE REVIEW

File: SPECS/td-agent/td-agent.spec [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+%attr(0755,td-agent,td-agent) %dir %{_localstatedir}/log/td-agent
+%attr(0755,td-agent,td-agent) %dir %{_localstatedir}/log/td-agent/buffer
+%attr(0755,td-agent,td-agent) %dir %{_sysconfdir}/td-agent
+%attr(0755,td-agent,td-agent) %dir %{_sysconfdir}/td-agent/plugin
+# NOTE: %{_tmpfilesdir} is available since CentOS 7
+%attr(0755,td-agent,td-agent) %dir /tmp/td-agent
+
+%changelog
+* Tue Dec 01 2020 Henry Li <lihl@microsoft.com> - 4.0.1-1
+- Initial CBL-Mariner Version
```
Comment (PawelWMS) [inline SPECS/td-agent/td-agent.spec:211]: It's `Original version for CBL-Mariner` for our own specs.

File: SPECS/td-agent/td-agent.patch [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+-download "cool.io", "1.6.0"
+-download 'serverengine', '2.2.1'
+-download "oj", "3.10.6"
+-download "async-http", "0.50.13"
+-download "http_parser.rb", "0.6.0"
+-download "yajl-ruby", "1.4.1"
+-download "sigdump", "0.2.4"
+-download "tzinfo", "2.0.2"
+-download "tzinfo-data", "1.2019.3"
++"/usr/src/mariner/SOURCES/00-bundler-2.1.4.gem"
```
Comment (PawelWMS) [inline SPECS/td-agent/td-agent.patch:153]: Would it be possible to use our sources server for the binaries instead? Some of them go into multiple MBs and we should avoid keeping larger files in a git repo. @schmittjoseph: do you think it's possible (without any huge changes) to make the tooling try and download all sources missing locally from the server? Or maybe that's already happening?

File: SPECS/td-agent/td-agent.patch [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+ 
+       desc "Install core_gems"
+       task :core_gems => [:"download:core_gems", :ruby] do
+-        install_gems(@download_task.files_core_gems)
++        cd(DOWNLOADS_DIR) do
++          ensure_directory("core_gems") do
++            cp("/usr/src/mariner/SOURCES/gemfile.tar.gz", ".")
++            sh("tar", "xvf", "gemfile.tar.gz", "-C", ".")
++            core_gem_path = "#{DOWNLOADS_DIR}/core_gems"
++            core_gems_files = ["#{core_gem_path}/00-bundler-2.1.4.gem",
```
Comment (jslobodzian) [inline SPECS/td-agent/td-agent.patch:66]: Why do we need these binaries?  Can we not build them ourselves?  What does Fedora or others do for this?

**Review by jslobodzian — COMMENTED** [review]
Let's discuss offline.

File: SPECS/td-agent/td-agent.patch (hunk 2 above)
Comment (schmittjoseph) [reply SPECS/td-agent/td-agent.patch:153]: That's already happening. Any and all sources not found locally for a given spec will be retrieved from the `SOURCE_URL`.

File: SPECS/td-agent/td-agent.spec (hunk 1 above)
Comment (henryli001) [reply SPECS/td-agent/td-agent.spec]: Fixed

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**